File: Flower.Data/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Flower.Data.Entities
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [MaxLength(50)]
        public string? Sku { get; set; }

        [Required(ErrorMessage = "Tên sản phẩm không được để trống")]
        [MaxLength(200)]
        public string Name { get; set; }

        public string? Description { get; set; }

        [MaxLength(300)]
        public string? Slug { get; set; }

        [Range(0, double.MaxValue)]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        public decimal? DiscountPrice { get; set; }

        public int StockQuantity { get; set; }

        public string? ImageUrl { get; set; }

        public int CategoryProductId { get; set; }

        [ForeignKey("CategoryProductId")]
        public virtual CategoryProduct? CategoryProduct { get; set; }

        public int ViewCount { get; set; }

        public int AddToCartCount { get; set; }

        public virtual ICollection<ProductVariant>? ProductVariants { get; set; }

        public bool IsActive { get; set; } = true;

        [MaxLength(500)]
        public string? FlowerMeaning { get; set; }

        [MaxLength(200)]
        public string? Origin { get; set; }

        public string? CareInstruction { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}

File: flower-admin.frontend/src/types/product.ts
export interface Product {
  id: number
  sku?: string
  name: string
  description?: string
  slug?: string
  price: number
  stockQuantity: number
  imageUrl?: string
  images: ProductImage[]
  productCategoryId: number
  productCategoryName?: string
  isActive: boolean
  flowerMeaning?: string
  origin?: string
  careInstruction?: string
  viewCount: number
  createdAt?: string
  variants?: ProductVariant[]
}

export interface ProductImage {
  id: number
  imageUrl: string
  sortOrder: number
}

export interface ProductVariant {
  id: number
  productId: number
  name: string
  price: number
  sku?: string
  isDefault: boolean
}

export interface CreateProductVariantRequest {
  name: string
  price: number
  sku?: string
  isDefault?: boolean
}

export interface UpdateProductVariantRequest extends CreateProductVariantRequest {
  id: number
}

export interface CreateProductRequest {
  name: string
  slug?: string
  sku?: string
  description?: string
  price: number
  stockQuantity: number
  productCategoryId: number
  imageUrl?: string
  isActive?: boolean
  flowerMeaning?: string
  origin?: string
  careInstruction?: string
  newImages?: string[]
}

export interface UpdateProductRequest extends CreateProductRequest {
  id: number
}

export interface PagedResponse<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}
